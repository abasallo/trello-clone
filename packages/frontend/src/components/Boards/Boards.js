import React from 'react'

import PropTypes from 'prop-types'

import AddIcon from '@mui/icons-material/Add'

import { Grid } from '@material-ui/core'

import { AddButton, Board, BoardsGrid, Container } from './Boards.styled.components'

const Boards = (props) => {
  return (
    <Container>
      <BoardsGrid container spacing={2}>
        {props.boards.map((board) => (
          <Grid item xs={12} sm={6} md={4} lg={3} key={board.id}>
            <Board>{board.name}</Board>
          </Grid>
        ))}
      </BoardsGrid>
      <AddButton size="medium" color="secondary" aria-label="add">
        <AddIcon />
      </AddButton>
    </Container>
  )
}

Boards.propTypes = {
  boards: PropTypes.object
}

export default Boards
