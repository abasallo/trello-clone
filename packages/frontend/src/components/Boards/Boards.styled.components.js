import styled from 'styled-components'

import { Card, Fab, Grid } from '@material-ui/core'

import MaterialUIContainer from '@material-ui/core/Container'

export const Container = styled(MaterialUIContainer)`
  display: flex;
  flex-direction: column;
`

export const BoardsGrid = styled(Grid)`
  flex-grow: 1;
`
export const Board = styled(Card)`
  padding: 10px;
  min-height: 100px;
  text-align: center;
`

export const AddButton = styled(Fab)`
  margin-left: auto;
  margin-top: 10px;
`
