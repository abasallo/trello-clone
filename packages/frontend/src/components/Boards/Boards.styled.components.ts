import styled from 'styled-components'

import { Card, CardActions, Fab, Grid } from '@material-ui/core'

import MaterialUIContainer from '@material-ui/core/Container'

export const Container = styled(MaterialUIContainer)`
  display: flex;
  flex-direction: column;
`

export const BoardsGrid = styled(Grid)`
  flex-grow: 1;
`
export const BoardComponent = styled(Card)`
  padding: 10px;
  margin: 10px;
  min-height: 100px;
  text-align: center;
  background-color: ${(props) => props.color || '#000000'};
`

export const BoardActions = styled(CardActions)`
  text-align: right;
`

export const AddButton = styled(Fab)`
  margin-left: auto;
  margin-top: 10px;
`
