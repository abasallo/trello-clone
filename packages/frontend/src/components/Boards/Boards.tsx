import React, {useState} from 'react'

import AddIcon from '@mui/icons-material/Add'
import DeleteIcon from '@mui/icons-material/Delete'
import EditIcon from '@mui/icons-material/Edit'

import {Chip, Grid, IconButton, TextField} from '@material-ui/core'

import {AddButton, BoardComponent, BoardActions, BoardsGrid, Container} from './Boards.styled.components'

import {useAppDispatch, useAppSelector} from '../../redux/hooks'
import {AppDispatch} from '../../redux/store'

import {addBoardAsyncThunk, deleteBoardAsyncThunk, updateBoardAsyncThunk} from '../../redux/thunks/boards.thunks'

import {Board} from 'trello-clone-shared/src/model/board.model'

const addButtonOnClick = (dispatch: AppDispatch): void => {
    dispatch(addBoardAsyncThunk())
}
const updateButtonOnClick = (dispatch: AppDispatch, board: Board): void => {
    dispatch(updateBoardAsyncThunk(board))
}

const deleteButtonOnClick = (dispatch: AppDispatch, id?: number): void => {
    if (id) dispatch(deleteBoardAsyncThunk(id))
}

const Boards = () => {

    const [editMode, setEditMode] = useState(false)

    const state = useAppSelector(state => state)
    const dispatch: AppDispatch = useAppDispatch()

    const boards: Board[] = state.boards || []

    return (
        <Container>
            <BoardsGrid container spacing={2}>
                {boards.map((board) => (
                    <Grid item xs={12} sm={6} md={4} lg={3} key={board.id}>
                        <BoardComponent color={board.colour}>
                            {
                                editMode
                                    ? <TextField id="standard-basic" label="Standard" variant="standard"
                                                 defaultValue={board.name}
                                                 onBlur={(e) => {
                                                     updateButtonOnClick(dispatch, {
                                                         id: board.id,
                                                         name: e.target.value
                                                     })
                                                     setEditMode(false)
                                                 }}
                                    />
                                    : <Chip label={board.name} variant="outlined" onClick={() => setEditMode(true)}/>
                            }

                            <BoardActions disableSpacing>
                                <IconButton aria-label="edit" onClick={() => setEditMode(true)}>
                                    <EditIcon/>
                                </IconButton>
                                <IconButton aria-label="delete" onClick={() => deleteButtonOnClick(dispatch, board.id)}>
                                    <DeleteIcon/>
                                </IconButton>
                            </BoardActions>
                        </BoardComponent>
                    </Grid>
                ))}
            </BoardsGrid>
            <AddButton size="medium" color="secondary" aria-label="add" onClick={() => addButtonOnClick(dispatch)}>
                <AddIcon/>
            </AddButton>
        </Container>
    )
}

export default Boards
