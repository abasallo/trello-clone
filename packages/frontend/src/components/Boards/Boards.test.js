import React from 'react'
import renderer from 'react-test-renderer'

import Boards from './Boards'

const initialBoards = [
	{
		name: 'Board 1'
	},
	{
		name: 'Board 2'
	},
	{
		name: 'Board 3'
	}
]

test('Renders properly', () => expect(renderer.create(<Boards boards={initialBoards} />).toJSON()).toMatchSnapshot())
