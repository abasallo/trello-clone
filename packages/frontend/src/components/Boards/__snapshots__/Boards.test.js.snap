// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders properly 1`] = `
<div
  className="MuiContainer-root sc-bcXHqe bRLOeJ MuiContainer-maxWidthLg"
>
  <div
    className="MuiGrid-root sc-gswNZR fsLIpU MuiGrid-container MuiGrid-spacing-xs-2"
  >
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6 MuiGrid-grid-md-4 MuiGrid-grid-lg-3"
    >
      <div
        className="MuiPaper-root MuiCard-root sc-dkrFOg iSngnb MuiPaper-elevation1 MuiPaper-rounded"
      >
        Board 1
      </div>
    </div>
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6 MuiGrid-grid-md-4 MuiGrid-grid-lg-3"
    >
      <div
        className="MuiPaper-root MuiCard-root sc-dkrFOg iSngnb MuiPaper-elevation1 MuiPaper-rounded"
      >
        Board 2
      </div>
    </div>
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6 MuiGrid-grid-md-4 MuiGrid-grid-lg-3"
    >
      <div
        className="MuiPaper-root MuiCard-root sc-dkrFOg iSngnb MuiPaper-elevation1 MuiPaper-rounded"
      >
        Board 3
      </div>
    </div>
  </div>
  <button
    aria-label="add"
    className="MuiButtonBase-root MuiFab-root sc-hLBbgP locPzZ MuiFab-sizeMedium MuiFab-secondary"
    disabled={false}
    onBlur={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
  >
    <span
      className="MuiFab-label"
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
        data-testid="AddIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
        />
      </svg>
    </span>
  </button>
</div>
`;
